src/python_testing/matter_testing_infrastructure/matter/testing/spec_parsing.py

```diff
@@ -441,6 +478,223 @@ def str_to_access_type(privilege_str: str) -> int:
 
         return (read_access, write_access, invoke_access)
 
+    def _parse_basic_field_attributes(self, xml_field: ElementTree.Element, component_type: DataTypeEnum, component_tags: dict) -> tuple[str, uint] | None:
+        """
+        Extract basic field attributes (name and ID) from XML element.
+
+        Args:
+            xml_field: XML element representing the field/item/bitfield
+            component_type: Type of component being parsed (struct/enum/bitmap)
+            component_tags: Mapping of component types to their XML tag info
+
+        Returns:
+            Tuple of (name, id) if successful, None if parsing failed
+        """
+        try:
+            name = xml_field.attrib['name']
+            id = uint(int(xml_field.attrib[component_tags[component_type].id_attrib], 0))
+            return (name, id)
+        except (KeyError, ValueError):
+            return None
+
+    def _determine_optional_status(self, xml_field: ElementTree.Element) -> bool:
+        """
+        Determine if a field is optional based on XML attributes and child elements.
+
+        Checks both:
+        1. isOptional attribute on the field element
+        2. Presence of optionalConform child element
+
+        Args:
+            xml_field: XML element representing the field
+
+        Returns:
+            True if field is optional, False otherwise
+        """
+        # Check isOptional attribute
+        if 'isOptional' in xml_field.attrib and xml_field.attrib['isOptional'] == 'true':
+            return True
+
+        # Check for optionalConform child element
+        optional_conform = xml_field.find('./optionalConform')
+        return optional_conform is not None
+
+    def _determine_nullable_status(self, xml_field: ElementTree.Element) -> bool:
+        """
+        Determine if a field is nullable based on XML attributes and child elements.
+
+        Checks both:
+        1. isNullable attribute on the field element
+        2. nullable attribute on quality child element
+
+        Args:
+            xml_field: XML element representing the field
+
+        Returns:
+            True if field is nullable, False otherwise
+        """
+        # Check isNullable attribute
+        if 'isNullable' in xml_field.attrib and xml_field.attrib['isNullable'] == 'true':
+            return True
+
+        # Check quality child element for nullable attribute
+        quality = xml_field.find('./quality')
+        if quality is not None and 'nullable' in quality.attrib and quality.attrib['nullable'] == 'true':
+            return True
+
+        return False
+
+    def _parse_field_constraints(self, xml_field: ElementTree.Element) -> dict | None:
```

Comment:
Could we explain why we return None instead of empty dictionary if no constraints?
I would maybe imagine that a dict is easier to handle than always separately consider None.